test/stdlib/symbol-visibility.test-sh

```diff
@@ -0,0 +1,19 @@
+// Ensure that we do not export any weak symbols from the dylibs.
+//
+// Weak symbols require additional work from the loader to resolve the symbol at
+// load time and can cause ODR violations as well as unexpected symbol
+// satisfaction because the weak symbol may be used from a separate module.
+
+// RUN: %empty-directory(%t)
+
+// RUN: %llvm-nm --defined-only --extern-only --demangle %platform-dylib-dir/%target-library-name(swiftCore) > %t/all.txt
+// RUN: %llvm-nm --defined-only --extern-only --no-weak --demangle %platform-dylib-dir/%target-library-name(swiftCore) > %t/no-weak.txt
+// RUN: diff %t/all.txt %t/no-weak.txt
+
+// RUN: %llvm-nm --defined-only --extern-only --demangle %platform-dylib-dir/%target-library-name(swiftRemoteMirror) > %t/all.txt
```

Comment:
To make using `diff -u` more effective, the name of the output files at line 9, 10, 13 and 14 should contain the name of the dynamic library being considered.